libs/sdk-ui-ext/src/internal/components/BaseVisualization.tsx

```diff
@@ -270,10 +257,39 @@ export class BaseVisualization extends React.PureComponent<IBaseVisualizationPro
         return !isEqual(omit(currentReferencePoint, "properties"), omit(nextReferencePoint, "properties"));
     }
 
+    private getVisualizationProps(): IVisProps {
+        return {
+            locale: this.props.locale,
+            dateFormat: this.props.dateFormat,
+            dimensions: {
+                width: this.props.width,
+                height: this.props.height,
+            },
+            custom: {
+                drillableItems: this.props.drillableItems,
+                totalsEditAllowed: this.props.totalsEditAllowed,
+            },
+            config: this.props.config,
+            theme: this.props.theme,
+            executionConfig: this.props.executionConfig,
+        };
+    }
+
     public getInsightWithDrillDownApplied(
         sourceVisualization: IInsight,
         drillDownContext: IDrillDownContext,
     ): IInsight {
         return this.visualization.getInsightWithDrillDownApplied(sourceVisualization, drillDownContext);
     }
+
+    public getExecution() {
+        this.setupVisualization(this.props);
+        this.updateVisualization();
+
+        return this.visualization.getExecution(
+            this.getVisualizationProps(),
+            this.props.insight,
+            this.executionFactory,
+        );
+    }
```

Comment:
isn't this triggering double-execution? i don't remember 100% but have a hazy recollection that vis.update() triggers internal state update and then re-render (and thus possibly also execution). i may be wrong.. please double-check.

if that's the case, please think about how to remedy the situation. it may be tricky due to messed up life-cycle on the update

Reply:
This is a really good point. After putting more thought in, there is no need to call `updateVisualization`, we just need to be sure that visualization is set up. Will update it accordingly